prereise/gather/flexibilitydata/geo/geo_data.py

```diff
@@ -0,0 +1,415 @@
+                # total population in this fips
+                total_pops = fips_pop_df.loc[fips_pop_df["fips"]==fipss[k],"population"]
+                total_pops = total_pops.to_list()[0]
+                if fipss[k] in all_fips:
+                    all_pops[all_fips.index(fipss[k])] += weights[k]
+                else:
+                    all_fips.append(fipss[k])
+                    all_pops.append(weights[k]*total_pops)
+                    
+        eiaid2fips.update({key: [all_fips, all_pops]})
+
+    with open(cache_path+"/eiaid2fips.pkl", 'wb') as fh:
+        pkl.dump(eiaid2fips, fh)
+    
+
+
+def fips_zip_conversion(raw_data_path, cache_path):
+    """Create a two-way mapping for all ZIP and FIPS in the crosswalk data
+        save to dictionary files for future use
+
+    param string raw_data_path: folder that contains downloaded raw data
+    param string cache_path: folder to store processed cache files
+    :return: (*None*)
+    """
+    
+    df_raw = pd.read_csv(raw_data_path+"/county_to_zip.csv")
+    all_fips = df_raw['county'].astype('int32')
+    all_zip = df_raw['zip'].astype('int32')
+    all_weights = df_raw['tot_ratio']
+
+    # create zip -> counties mapping
+    zip2fips = {}
+    
+    for i in pd.unique(all_zip):
+        idx = all_zip.index[all_zip == i]
+        
+        cty = all_fips[idx].tolist()
+        wgt = all_weights[idx].tolist()
+        
+        zip2fips.update({i: (cty, wgt)})
+
+    with open(cache_path+"/zip2fips.pkl", 'wb') as fh:
+        pkl.dump(zip2fips, fh)
+
+    # create county -> zips mapping
+    fip2zips = {}
+    
+    for i in pd.unique(all_fips):
+        idx = all_fips.index[all_fips == i]
+        
+        zips = all_zip[idx].tolist()
+        wgt = all_weights[idx].tolist()
+        
+        fip2zips.update({i: (zips, wgt)})
+
+
+    with open(cache_path+"/fip2zips.pkl", 'wb') as fh:
+        pkl.dump(fip2zips, fh)
+
+
+def get_fips_population(raw_data_path, cache_path):
+    """Match county population and county FIPS data to produce concise FIPS population
+        save to a dictonary in cache folder with key being 5-digit FIPS codes.
+
+    param string raw_data_path: folder that contains downloaded raw data
+    param string cache_path: folder to store processed cache files
+    :return: (*None*)
+    """
+
+    cty_pop_df = pd.read_csv(raw_data_path+"/county_population.csv", encoding='cp1252')
+    cty_name_df = pd.read_csv(raw_data_path+"/county_fips_master.csv", encoding='cp1252')
+
+    pops = np.zeros(len(cty_name_df.index))
+    for cty in cty_name_df.index:
+        name = cty_name_df['county_name'][cty]
+        state = cty_name_df['state_name'][cty]
+
+        tmp = cty_pop_df.loc[(cty_pop_df["CTYNAME"]==name) & (cty_pop_df["STNAME"]==state), "POPESTIMATE2019"]
+        # if no data
+        if len(tmp) == 0:
+            pops[cty] = 0
+        elif len(tmp) == 1:
+            pops[cty] = tmp
+        else:
+            pops[cty] = tmp.to_list()[0]
```

Comment:
What are the other values that might be listed if `len(tmp) > 1`?

Reply:
There is one unique case where a fips and name is duplicated in the original data (I think is D.C.), so this is just to take the 1st data for that one.